src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -197,13 +197,12 @@ public function request(string $method, string $url, array $options = []): Respo
             $curlopts[\CURLOPT_RESOLVE] = $resolve;
         }
 
+        $curlopts[\CURLOPT_CUSTOMREQUEST] = $method;
         if ('POST' === $method) {
             // Use CURLOPT_POST to have browser-like POST-to-GET redirects for 301, 302 and 303
```

Comment:
@nicolas-grekas wouldn't the usage of `CURLOPT_CUSTOMREQUEST` defeat this comment ?

Reply:
I think it's fine because it's set after.

Reply:
I, too, thought that, but after testing it just now it seems like it might not be fine...

Reply:
I looked into this, and it seems like this would be an issue in plain cURL, but actually works correctly in Symfony because redirects are handled in `CurlResponse::parseHeaderLine` instead of being handled by cURL itself.
I might still make a PR to add a test for this.